Biasing/src/PhotoNuclearXsecBiasingOperator.cxx

```diff
@@ -0,0 +1,97 @@
+        XsecBiasingOperator(name) {
+    }
+
+    PhotoNuclearXsecBiasingOperator::~PhotoNuclearXsecBiasingOperator() {
+    }
+
+    G4VBiasingOperation* PhotoNuclearXsecBiasingOperator::ProposeOccurenceBiasingOperation(
+            const G4Track* track, const G4BiasingProcessInterface* callingProcess) {
+    
+        /*std::cout << "[ PhotoNuclearXsecBiasingOperator ]: "
+                  << "Parent ID: " << track->GetParentID() 
+                  << " Created within " << track->GetLogicalVolumeAtVertex()->GetName() 
+                  << std::endl;*/
+
+        if (track->GetParentID() != 1) return 0; 
+
+        std::cout << "[ PhotoNuclearXsecBiasingOperator ]: " 
+                  << "Kinetic energy: " << track->GetKineticEnergy() 
+                  << " MeV" << std::endl;
+
+        if (track->GetKineticEnergy() < XsecBiasingOperator::threshold_) return 0; 
+
+        std::cout << "[ PhotoNuclearXsecBiasingOperator ]: " 
+                  << "Calling process: " 
+                  << callingProcess->GetWrappedProcess()->GetProcessName() 
+                  << std::endl;
+
+        std::string currentProcess = callingProcess->GetWrappedProcess()->GetProcessName(); 
+        if (currentProcess.compare(this->getProcessToBias()) == 0) { 
+            
+            G4double interactionLength = callingProcess->GetWrappedProcess()->GetCurrentInteractionLength();
+            std::cout << "[ PhotoNuclearXsecBiasingOperator ]: "
+                      << "PN Interaction length: " 
+                      << interactionLength << std::endl;
+
+            pnXsecUnbiased_ = 1./interactionLength;
+            std::cout << "[ PhotoNuclearXsecBiasingOperator ]: Unbiased PN xsec: "
+                      << pnXsecUnbiased_ << std::endl;
+
+            pnXsecBiased_ = pnXsecUnbiased_*xsecFactor_; 
+            std::cout << "[ PhotoNuclearXsecBiasingOperator ]: Biased PN xsec: "
+                      << pnXsecBiased_ << std::endl;
+
+            xsecOperation->SetBiasedCrossSection(pnXsecBiased_);
+            xsecOperation->Sample();
+
+            return xsecOperation;
+    
+        } else if (currentProcess.compare(CONVERSION_PROCESS) == 0) { 
+            
+            G4double interactionLength = callingProcess->GetWrappedProcess()->GetCurrentInteractionLength();
+            /*std::cout << "[ PhotoNuclearXsecBiasingOperator ]: "
+                      << "EM Interaction length: " 
+                      << interactionLength << std::endl;*/
+
+            double emXsecUnbiased = 1./interactionLength; 
+            /*std::cout << "[ PhotoNuclearXsecBiasingOperator ]: Unbiased EM xsec: "
+                      << emXsecUnbiased << std::endl;*/
+
+            double emXsecBiased = std::max(emXsecUnbiased + pnXsecUnbiased_ - pnXsecBiased_, pnXsecUnbiased_);  
+            /*std::cout << "[ PhotoNuclearXsecBiasingOperator ]: Biased EM xsec: "
+                      << emXsecBiased << std::endl;*/
+
```

Comment:
May be a good idea to write a warning message here if emXsecBiased==pnXsecUnbiased that the biasing factor is too large.

Reply:
Done.